distributed/tests/test_scheduler.py

```diff
@@ -737,6 +737,31 @@ async def test_retire_workers_empty(s):
     await s.retire_workers(workers=[])
 
 
+@gen_cluster(client=True, nthreads=[("", 1)] * 10)
+async def test_retire_workers_concurrently(c, s, *workers):
+    """Test that replicate(), rebalance(), and retire_workers() can be called at the
+    same time. Note how these three activities are mutually exclusive with each other,
+    but you can have multiple instances of retire_workers running at the same time.
+
+    See also
+    --------
+    test_active_memory_manager.py::test_RetireWorker_stress
+    """
+    futs = c.map(inc, range(100))
+    await wait(futs)
+    before = s.transition_counter
+    await asyncio.gather(
+        *(
+            s.replicate(keys=[fut.key for fut in futs], n=2),
+            s.rebalance(),
+            s.retire_workers(workers=[workers[0].address]),
+            s.retire_workers(workers=[workers[1].address]),
+        )
+    )
+    await c.gather(futs)
+    assert s.transition_counter == before
```

Comment:
This test doesn't care whether the three run concurrently or sequentially, does it? In fact, they could be replaced with no-ops and this test would pass.

I would like to see a test that actually verifies that the two retire_workers are indeed running concurrently.